### DISCUSSION TITLE
how to use your code to design a Z drive coil?

### DISCUSSION THREAD

**Discussion by Zhaoanping** [gBringout/CoilDesign#9] [General]
Dear Gael,

By running your code I have found that your code is quite useful; and here I would like to thank you for posting your excellent work at the GitHub!

I have been running your default code "TransMag_Drive" to get the driving coil in Y direction; and I also have modified your code a bit to get satisfactionary driving coil in X direction. However, I could not get driving coil in Z direction. Could you please advise me how the Z direction drving coil can be obtained with your code?    

BRs

Anping

**Comment by gBringout** [comment]
Dear Anping,

thanks you for the question.

I am still investigating the Z Coil. I suppose that the wire extractor does not like this type of coil.

Regarding the question you send me per mail on the efficiency:
![image](https://user-images.githubusercontent.com/5993858/110159158-9fb21700-7dea-11eb-8645-0a40619684bf.png)
->How to get the 51.2 uT/A?.

Using the schimdt semi-normalisation und the standard model
![image](https://user-images.githubusercontent.com/5993858/110159669-4d252a80-7deb-11eb-8815-3e920eb29e70.png)
, you get those numbers:
![image](https://user-images.githubusercontent.com/5993858/110159737-6037fa80-7deb-11eb-8be5-251b43dcc003.png)
i.e. an efficiency of around 44.6 uT/A

Closing the wire loops get you an slightly higher efficiency:
![image](https://user-images.githubusercontent.com/5993858/110159402-f1f33800-7dea-11eb-8e39-b455ade406be.png)

![image](https://user-images.githubusercontent.com/5993858/110159293-ce2ff200-7dea-11eb-89a1-996abff278ee.png)

I quickly closed the loop with these code:
```
%wiresaved = wire;
%close the wire loop
for i=1:size(wire,2)
    wire(i).Coord(:,end+1) = wire(i).Coord(:,1);
end
```
It is still note the 51 published. I will try to also have a look at it.

**Comment by Zhaoanping** [comment]
Dear Gael,

Thank you very much for your reply, which certainly lets me understand your code more.

Look forward to hearing from you about the Z drive coil implementation. In addition, regarding to this, I think that (unlike the X- and Y-direction driving coil), in the simulation setup, we first have to set the two parts of the z-direction driving coils have the same current direction and then do the wire extraction. Actually, the Z direction driving coil should be similar to the Helmholtz coil. If these two parts of the Z-direction coils have the opposite current direction, then we could not get the right Z-direction coils (here right means that we should get homogeneous B field in the target region); and I used two simple Helmholtz coils (each has one turn) to demonstrate it, please see the two figures below for detail.

[cid:image005.png@01D71413.73D5BB00]


[cid:image006.png@01D71413.73D5BB00]

BRs

Anping

发件人: Gael Bringout <notifications@github.com>
发送时间: 2021年3月6日 2:49
收件人: gBringout/CoilDesign <CoilDesign@noreply.github.com>
抄送: Anping ZHAO（赵安平） <zhaoanping@huami.com>; Author <author@noreply.github.com>
主题: Re: [gBringout/CoilDesign] how to use your code to design a Z drive coil? (#9)


Dear Anping,

thanks you for the question.

I am still investigating the Z Coil. I suppose that the wire extractor does not like this type of coil.

Regarding the question you send me per mail on the efficiency:
[image]<https://user-images.githubusercontent.com/5993858/110159158-9fb21700-7dea-11eb-8645-0a40619684bf.png>
->How to get the 51.2 uT/A?.

Using the schimdt semi-normalisation und the standard model
[image]<https://user-images.githubusercontent.com/5993858/110159669-4d252a80-7deb-11eb-8815-3e920eb29e70.png>
, you get those numbers:
[image]<https://user-images.githubusercontent.com/5993858/110159737-6037fa80-7deb-11eb-8be5-251b43dcc003.png>
i.e. an efficiency of around 44.6 uT/A

Closing the wire loops get you an slightly higher efficiency:
[image]<https://user-images.githubusercontent.com/5993858/110159402-f1f33800-7dea-11eb-8e39-b455ade406be.png>

[image]<https://user-images.githubusercontent.com/5993858/110159293-ce2ff200-7dea-11eb-89a1-996abff278ee.png>

I quickly closed the loop with these code:

%wiresaved = wire;

%close the wire loop

for i=1:size(wire,2)

    wire(i).Coord(:,end+1) = wire(i).Coord(:,1);

end

It is still note the 51 published. I will try to also have a look at it.

**Reply by gBringout** [reply]
Dear Anping,

I agree with you on the point that the coild should be similar to an Helmoltz one: generating an homogeneous field Bz in the middle.
The optimization script is also producing such coils. 

The question/issue here is more: how do you define the direction of something in a 3D coordinate system.

How do you go from the direction/amplitude represented by the stream function to the direction of the flowing current.

Their is their either a bug or a representation issue.

PS: your picture are someone not displayed by me.

Regards,
Gael

**Comment by Zhaoanping** [comment]
Dear Gael,

In order to ensure that you can display the picture, I send it to you again in an attached file. In slide 3, I also add more questions.

Wish the problem related to the Z-coil could be solved soon.

Have a nice weekend!

BRs

Anping

发件人: Gael Bringout ***@***.***>
发送时间: 2021年3月12日 0:31
收件人: gBringout/CoilDesign ***@***.***>
抄送: Anping ZHAO（赵安平） ***@***.***>; Author ***@***.***>
主题: Re: [gBringout/CoilDesign] how to use your code to design a Z drive coil? (#9)


Dear Anping,

I agree with you on the point that the coild should be similar to an Helmoltz one: generating an homogeneous field Bz in the middle.
The optimization script is also producing such coils.

The question/issue here is more: how do you define the direction of something in a 3D coordinate system.

How do you go from the direction/amplitude represented by the stream function to the direction of the flowing current.

Their is their either a bug or a representation issue.

PS: your picture are someone not displayed by me.

Regards,
Gael

**Comment by Zhaoanping** [comment]
Dear Gael,

I just would like to let you know that the ending of the email address of my company has been changed from “huami.com” to “zepp.com”, which means my current email address is ***@***.***

BRs

Anping

发件人: Anping ZHAO（赵安平）
发送时间: 2021年3月12日 11:08
收件人: 'gBringout/CoilDesign' ***@***.***>; gBringout/CoilDesign ***@***.***>
抄送: Author ***@***.***>; ***@***.***' ***@***.***>
主题: 答复: [gBringout/CoilDesign] how to use your code to design a Z drive coil? (#9)

Dear Gael,

In order to ensure that you can display the picture, I send it to you again in an attached file. In slide 3, I also add more questions.

Wish the problem related to the Z-coil could be solved soon.

Have a nice weekend!

BRs

Anping

发件人: Gael Bringout ***@***.******@***.***>>
发送时间: 2021年3月12日 0:31
收件人: gBringout/CoilDesign ***@***.******@***.***>>
抄送: Anping ZHAO（赵安平） ***@***.******@***.***>>; Author ***@***.******@***.***>>
主题: Re: [gBringout/CoilDesign] how to use your code to design a Z drive coil? (#9)


Dear Anping,

I agree with you on the point that the coild should be similar to an Helmoltz one: generating an homogeneous field Bz in the middle.
The optimization script is also producing such coils.

The question/issue here is more: how do you define the direction of something in a 3D coordinate system.

How do you go from the direction/amplitude represented by the stream function to the direction of the flowing current.

Their is their either a bug or a representation issue.

PS: your picture are someone not displayed by me.

Regards,
Gael

**Comment by gBringout** [comment]
Dear Anping,

please use the Github interface to write back and join files. The interface is apparently not able to process Mails correctly. Your attachment are also somehow not attached.

**Efficiency**
I looked further into the efficiency discrepancy. I did not reach again the published 51.2 uT/A, but came close at 51.08 uT/A.

To do so, the wire path as to by further optimized and corrected, I also have integrated some connection correction. Note that the modification are done in different software, and the coil ended up rotated (or became a X-Drive coil).
It looks like that:
![image](https://user-images.githubusercontent.com/5993858/113538325-3a189c80-95db-11eb-88cf-233945a73459.png)

**Z Gradient Coil**
I have not dived again in the math, but I can confirm that you have to be cautious with current direction and the interpretation it. The way you are referring to is the correct one: always go back to well known coil design to make sure the current direction set for the simulation is the intended one. I did not had the need to make this part 100% watertight and easy to use, as the design obtained from the script will always be heavily modified in other software and imported back and simulated to be checked. It may be possible to numerically optimized each of those steps, but for my application it was more practical to do it this way.

** Application related to your question **
You are mentioning your company. I suppose you are pursuing a commercial Goal, am I correct? Did you checked the license of this software package?

**Comment by Zhaoanping** [comment]
Dear Gael,

Thanks for your reply!

I agree with you that the coil efficiency will be affected by different optimization setup and the way to connect different parts of the wire.
In addition, it is true that derive Z direction coil will need a lot of work and none of the public code can work for a MRI system. We recently have hired a fresh PhD student (offer has been sent out) to write the code for coils design and we believe that he can complete the work very quick as his PhD work was focusing on the coil designs.

Thanks a lot once again for the suggestion/discussion you have given to me.

BRs

Anping

发件人: Gael Bringout ***@***.***>
发送时间: 2021年4月5日 12:58
收件人: gBringout/CoilDesign ***@***.***>
抄送: Anping ZHAO（赵安平） ***@***.***>; Author ***@***.***>
主题: Re: [gBringout/CoilDesign] how to use your code to design a Z drive coil? (#9)


Dear Anping,

please use the Github interface to write back and join files. The interface is apparently not able to process Mails correctly. Your attachment are also somehow not attached.

Efficiency
I looked further into the efficiency discrepancy. I did not reach again the published 51.2 uT/A, but came close at 51.08 uT/A.

To do so, the wire path as to by further optimized and corrected, I also have integrated some connection correction. Note that the modification are done in different software, and the coil ended up rotated (or became a X-Drive coil).
It looks like that:
[image]<https://user-images.githubusercontent.com/5993858/113538325-3a189c80-95db-11eb-88cf-233945a73459.png>

Z Gradient Coil
I have not dived again in the math, but I can confirm that you have to be cautious with current direction and the interpretation it. The way you are referring to is the correct one: always go back to well known coil design to make sure the current direction set for the simulation is the intended one. I did not had the need to make this part 100% watertight and easy to use, as the design obtained from the script will always be heavily modified in other software and imported back and simulated to be checked. It may be possible to numerically optimized each of those steps, but for my application it was more practical to do it this way.

** Application related to your question **
You are mentioning your company. I suppose you are pursuing a commercial Goal, am I correct? Did you checked the license of this software package?

**Reply by gBringout** [reply]
Hi Anping,

Nice to see that someone will dive into this interesting topic!

The code is indeed able to make coils for MRI, but the post-processing of the raw design has still to be done. This is true for MPI and MRI coils.

Have fun with your design!

Best regards,
Gael
